Comp2023/src/main/java/frc/robot/subsystems/Swerve.java

```diff
@@ -538,6 +542,70 @@ public void drive(Translation2d translation, double rotation, boolean fieldRelat
     }
   }
 
+  public void driveBalance( )
+  {
+    pitch = m_pigeon.getUnadjustedPitch( ).getWPIRotation2d( ).getDegrees( );
+    DataLogManager.log("Balance Off By: " + pitch);
+
+    if (Math.abs(pitch) > 5)
+    {
+      timer.reset( );
+      while (Math.abs(pitch) > 5 || !(timer.hasElapsed(5)))
```

Comment:
While loops are a no-no in WPILib programming. This loop has to be written to be called from the Command and give up control during each pass.

Reply:
I am not sure how to fix this. So do I need to make it into a different loop?

Reply:
The driveBalance command IS your loop. It will call m_swerve.driveBalanceExecute() to do the driving. As long as driveBalance doesn't have its isFInished method become true (which you have using a timer), then m_swerve.driveBalanceExecute will keep getting called and can do all the driving for you. We can discuss tomorrow (or next time you're in), but the method in the swerve drive really just needs do the correct driving with no timers, waits, or whiles. Those are managed by the command that calls it. Look at the method I wrote up--it's pretty close to all of the driving decisions you need. The robot either drives forward, reverse, or stops the motors based on the pigeon.